frontend/src/components/metadata/CreateMetadataDefinition.tsx

```diff
@@ -0,0 +1,307 @@
+						return (<Step key={index}>
+								{index == 0 ? 
+									<StepLabel>Add metadata entry*
+										{index == activeStep - 1 ? 
+										<IconButton color="primary" size="small" onClick={() => addNewField(index)}>
+											<AddBoxIcon />
+										</IconButton>
+										: <></>}
+									</StepLabel>
+								:
+									<StepLabel>Add additional entry
+										{index == activeStep - 1 ? 
+										<>
+											<IconButton color="primary" size="small" onClick={() => addNewField(index)}>
+												<AddBoxIcon />
+											</IconButton>
+											<IconButton color="primary" size="small" onClick={() => removeField(index)}>
+												<DeleteOutlineIcon />
+											</IconButton>
+										</>
+										: 
+										<IconButton size="small" onClick={() => removeField(index)}>
+											<DeleteOutlineIcon />
+										</IconButton>}
+									</StepLabel>
+								}
+								<StepContent>
+									<Grid container>
+										<Grid xs={4} md={4}>
+											<FormGroup>
+												<FormControlLabel control={<Checkbox 
+													checked={input.list}
+													onChange={(event) => { handleInputChange(index, "list", event.target.value); }}
+													/>} label="Contains List" />
+												<FormControlLabel control={<Checkbox 
+													checked={input.required}
+													onChange={(event) => { handleInputChange(index, "required", event.target.value); }}
+													/>} label="Required" />
+											</FormGroup>
+										</Grid>
+									</Grid>
+									<TextField
+										variant="outlined"
+										margin="normal"
+										required
+										fullWidth
+										id="field-name"
+										label="Field Name"
+										placeholder="Please enter field name"
+										value={input.name}
+										onChange={(event) => { handleInputChange(index, "name", event.target.value); }}
+									/>
+									<FormControl fullWidth>
+										<InputLabel id="widget-type-select">Select Widget Type</InputLabel>
+										<Select
+											labelId="widget-type-select-label"
+											id="widget-type-select-label"
+											value={input.widgetType}
+											onChange={(event) => { handleInputChange(index, "widgetType", event.target.value); }}
+											label="widgetType">
+											{widgetTypes.map((input, index) => {
+												return (
+													<MenuItem value={input}>{input}</MenuItem>
+												);
+											})}
+										</Select>
+									</FormControl>
+									<TextField
+										variant="outlined"
+										margin="normal"
+										required
+										fullWidth
+										id="datatype"
+										label="Field Data Type"
+										placeholder="Please enter field data type (e.g. int, str, enum)"
+										name="Field Data Type"
+										value={input.config.type}
+										onChange={(event) => { handleInputChange(index, "type", event.target.value); }}
+									/>
+									{(input.config.type == "enum") ? <>
+										<TextField
```

Comment:
If you plan to address the multiple options of Enum in the next PR please create a task and add a TODO here :-)

Reply:
Added To-dos wherever applicable